### DISCUSSION TITLE
Help for supporting Atlantic thermostat devices

### DISCUSSION THREAD

**Discussion by TTR6214** [Koenkk/zigbee2mqtt#16807] [Devices]
I have an Atlantic thermostat, called Navilink 128, composed of
- an interface connected to the boiler (heat and hot water), acting as coordinator or relay,
- a remote control with temperature reading and a display for setting, acting as end device.

These two equipments communicate by the Zigbee protocol.
In the user manual, the manufacturer clearly states that the equipment can be paired to an existing Zigbee network (see pp.38 in french).
[Notice.pdf](https://github.com/Koenkk/zigbee2mqtt/files/10823424/Notice.pdf)

I managed to sniff the exchanges between the interface and the remote control (see attachment).
[sniff_wireshark.zip](https://github.com/Koenkk/zigbee2mqtt/files/10823200/sniff_wireshark.zip)

Then I tried to add these two devices to my existing Zigbee network.
The interview process is a success for the interface.
However the interview procedure is random for the remote control. Sometimes it works, sometimes the equipment leaves the Zigbee network, sometimes it does not finish.

At this level, I assume that the remote control must be bind with the interface during the interview procedure.
You can see the file "pairing_remote_controller_with_interface.pcapng" for the pairing process between the interface and the remote controller.

Now I try to support the first device, i.e. the interface, in Zigbee2MQTT.
**This is my first time trying to support a new device in Zigbee2MQTT.
That's why I need help, because I feel like I'm not starting with simple equipment.**
Below the external converter that I try to create.

```
const fz = require('zigbee-herdsman-converters/converters/fromZigbee');
const tz = require('zigbee-herdsman-converters/converters/toZigbee');
const exposes = require('zigbee-herdsman-converters/lib/exposes');
const reporting = require('zigbee-herdsman-converters/lib/reporting');
const extend = require('zigbee-herdsman-converters/lib/extend');
const e = exposes.presets;
const ea = exposes.access;

const atlanticManufacturer = {manufacturerCode: 0x125b};

module.exports = [
    {
    zigbeeModel: ['NAVILINK SIF'],
    model: 'Interface Navilink',
    vendor: 'Atlantic',
    description: 'Interface Navilink for thermostat 125/128',
        fromZigbee: [
        	fz.thermostat,
        	fz.thermostat_weekly_schedule
        ],
        toZigbee: [
        	tz.thermostat_local_temperature,
        	tz.thermostat_system_mode,
        	tz.thermostat_running_state,
        	tz.thermostat_occupied_heating_setpoint,
        	tz.thermostat_control_sequence_of_operation,
        	tz.thermostat_weekly_schedule,
        	tz.thermostat_clear_weekly_schedule,
        	tz.thermostat_temperature_setpoint_hold,
        	tz.thermostat_temperature_setpoint_hold_duration
        ],
        meta: {disableDefaultResponse: true, multiEndpoint: true},
        configure: async (device, coordinatorEndpoint, logger) => {
            const heatEndpoint = device.getEndpoint(1);
            const waterEndpoint = device.getEndpoint(3);
            const temperatureEndpoint = device.getEndpoint(2);
            const binds = [
                'genBasic',
                'genIdentify',
                'genAlarms',
                'genTime',
                'hvacThermostat',
            ];
            await reporting.bind(heatEndpoint, coordinatorEndpoint, binds);
            await reporting.thermostatTemperature(heatEndpoint);
            await reporting.thermostatRunningMode(heatEndpoint);
            await reporting.thermostatSystemMode(heatEndpoint);
            await reporting.thermostatOccupiedCoolingSetpoint(heatEndpoint);
            await reporting.thermostatOccupiedHeatingSetpoint(heatEndpoint);
            await reporting.thermostatTemperatureSetpointHold(heatEndpoint);
            await reporting.thermostatTemperatureSetpointHoldDuration(heatEndpoint);
        },
        exposes: [
            exposes.climate()
                .withLocalTemperature()
                .withSetpoint('occupied_heating_setpoint', 5, 32, 0.5)
                .withSystemMode(['off', 'auto', 'heat']),
            exposes.binary('temperature_setpoint_hold', ea.ALL, true, false)
                .withDescription('Prevent changes. `false` = run normally. `true` = prevent from making changes. Must be set to `false` when system_mode = off or `true` for heat'),
            exposes.numeric('temperature_setpoint_hold_duration', ea.ALL)
                .withValueMin(0)
                .withValueMax(65535)
                .withDescription('Period in minutes for which the setpoint hold will be active. 65535 = attribute not used. 0 to 360 to match the remote display'),
        ],
    },
    {
        zigbeeModel: ['NAVILINK 128\u0000\u0000\u0000 2.0.8\u0000\u0000PRT_Clust'],
        model: 'Navilink 128',
        vendor: 'Atlantic',
        description: 'Heating thermostat remote control',
        fromZigbee: [fz.temperature, fz.battery],
        toZigbee: [],
        exposes: [e.battery(), e.temperature()],
        configure: async (device, coordinatorEndpoint, logger) => {
            const endpoint1 = device.getEndpoint(1);
            const endpoint2 = device.getEndpoint(2);
            const endpoint3 = device.getEndpoint(3);
            const endpoint232 = device.getEndpoint(232);
            
            await reporting.bind(endpoint1, coordinatorEndpoint, [
            	'genBasic',
            	'genIdentify',
            	'genTime',
                //'genPowerCfg',
            	'hvacThermostat',
            	//'msTemperatureMeasurement',
            	//'manuSpecificUbisysDimmerSetup'
            ]);
            //await reporting.bind(endpoint2, coordinatorEndpoint, [
                //'msTemperatureMeasurement'
            //]);
            //await reporting.bind(endpoint3, coordinatorEndpoint, [
                //'genTime',
                //'genPowerCfg'
            //]);
            //await reporting.bind(endpoint232, coordinatorEndpoint, [
                //'genPowerCfg'
            //]);
            
            
            await reporting.batteryPercentageRemaining(endpoint3);
            
            await reporting.temperature(endpoint2);
        },
    },
];
```

I did some research on the subject:

- another user (@Noneangel) has already supported an Atlantic Group device. Visually the interface equipment is identical. In his device.js file, the manufacturer code is identical (i.e., {manufacturerCode: 0x125b}).
However, I have difficulties to transpose its code to my use case.
https://github.com/Koenkk/zigbee-herdsman-converters/pull/4560

- recently a user supported a thermostat from another manufacturer.
https://github.com/Koenkk/zigbee2mqtt/issues/14926

- another thermostat is also supported by zigbee2MQTT. It is interesting because it is also composed of an interface and a remote control
https://www.zigbee2mqtt.io/devices/SLR2.html

I tried to use these three sources to build my extern converter, with some sucess and lots of error.

**Comment by Noneangel** [comment]
You were misled by the image. The device for which I added support is inside an interior unit (like the one in [this page](https://www.atlantic.fr/Climatiser-le-logement/Climatisation-reversible-pour-une-seule-piece/Les-climatiseurs-muraux/Muraux-DOJO?ref=8738260)). It is the device on the left of the hub on [this page](https://www.atlantic.fr/Piloter-les-appareils/Via-smartphone/Solutions-connectees/Pack-Naviclim). The device on the right is the Atlantic  hub (I don't use it as it is replaced by zigbe2mqtt).

**Comment by sylvaindd** [comment]
Hi @TTR6214 ,did you manage to get it to work ?
I also have the Navilink 128 and would be interested in this.

**Reply by TTR6214** [reply]
The answer is no, and I'm waiting for the winter to launch again my heat system and investigate the integration with Z2M ;-).

Feel free to help or propose a preliminary working configuration file.

**Reply by sylvaindd** [reply]
Unfortunately I didn't even manage to get it to pair.
I did some reverse engineering on the Cozytouch app to have some sensors and controls on HA, but a local solution would be better.

**Comment by TTR6214** [comment]
I found some documentation on the Navilink Interface : https://csa-iot.org/csa_product/navilink-interface/

After some research, it exist a Schneider thermostat with the same approach, i.e. interface and remote sensor : https://www.zigbee2mqtt.io/devices/CCTFR6700.html and https://www.zigbee2mqtt.io/devices/CCTFR6400.html

**Comment by gduteil** [comment]
I have the same thermostat (Navilink 128) with a Naema 2 Micro 25, I'm currently trying another approach using an OpenTherm gateway between the thermostat receiver and the boiler (https://diyless.com/product/esp8266-opentherm-gateway with a NodeMCU board).
Communication starts to work, but is unstable and gets corrupted, I suspect a problem the OpenTherm library when using 2 instances, I will need to look at it.

This is probably not the good place to talk about it as it's not related to zigbee2mqtt, but it's still referring to the same hardware, I will follow your progress and can keep you informed on mine if you are interested

**Reply by gduteil** [reply]
I tried few implementations of OpenTherm master or gateway, and none of them were satisfying.
I'm finally starting my own based on https://github.com/arthurrump/esphome-opentherm and https://github.com/ihormelnyk/opentherm_library.
After analyzing the signals, it appears the Navilink sends an invalid signal from time to time which makes the OpenTherm library ignoring the next message and/or considering most of the messages invalid.
I managed to fix that by filtering interrupts, all messages are received and validity is more reliable, nevertheless I still get every few hours an invalid value for some messages, I don't know if it's a transmission issue or another problem in interrupt handling, will need to investigate.

Remaining work will be :
  - Fix the remaining occasional invalid values
  - Create sensors for all messages
  - Add controls/overrides in HA

I will make a repo for all this in few weeks.

**Reply by gduteil** [reply]
I just made the repo and a short documentation : https://github.com/gduteil/OpenThermGateway/

**Comment by gduteil** [comment]
Any progress on your side ?

**Reply by gduteil** [reply]
Update from my side :

The OpenThermGateway has been working fine for the past months, communication is stable and most of the sensors are connected.
I did some tests for overrides but unfortunately I didn't find any way to control the boiler and still use the Navilink Thermstat using only the OpenTherm interface.
I don't have a Zigbee dongle, I ordered one but I haven't received it yet, so I check the Cozytouch protocol that was not working with the official integration (Overkiz). The protocol is totally different and I think it won't be wise to try to integrate it Overkiz, so I started a new custom integration.
I managed to get values from Cozytouch servers and control some parameters (central heating, domestic hot water, requested temperature), lot of parameters are also connected but I didn't have time to try to get a match.
I will upload the integration on my github, if you are interested in testing it please let me know.

Once this integration will be ready, I'll probably try to use the Zigbee to avoid relying on a Cloud service.

**Reply by Kinxkinx** [reply]
> Update from my side :
> I don't have a Zigbee dongle, I ordered one but I haven't received it yet, so I check the Cozytouch protocol that was not working with the official integration (Overkiz). 

Have you ever tried to use your  Zigbee dongle with this zinzin? I have a naviink 128 with the naema 25... But I can't connect the navilink to HA with Z2M using the Sonoff dongle...

**Comment by ggoo888** [comment]
Any updates on this?  I have a Amana PTAC and an Eden thermostat. It also says it is using Zigbee.  I'm wondering how to connect it to zigbee2mqtt.  I can't even get it to pair.  Are you guys able to get it to pair?

**Comment by cweiland** [comment]
Any news here?
